### PR TITLE
[Merged by Bors] - feat: Port/Data.QPF.Multivariate.Constructions.Fix

### PR DESCRIPTION
**Pull request by arienmalec**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by casavaca** [comment]
Pushed a small change to make it build.

File: Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    (∀ x, WEquiv (f₀ x) (f₁ x)) → WEquiv (q.P.wMk a f' f₀) (q.P.wMk a f' f₁)
+  | abs (a₀ : q.P.A) (f'₀ : q.P.drop.B a₀ ⟹ α) (f₀ : q.P.last.B a₀ → q.P.W α) (a₁ : q.P.A)
+    (f'₁ : q.P.drop.B a₁ ⟹ α) (f₁ : q.P.last.B a₁ → q.P.W α) :
+    abs ⟨a₀, q.P.appendContents f'₀ f₀⟩ = abs ⟨a₁, q.P.appendContents f'₁ f₁⟩ →
+      WEquiv (q.P.wMk a₀ f'₀ f₀) (q.P.wMk a₁ f'₁ f₁)
+  | trans (u v w : q.P.W α) : WEquiv u v → WEquiv v w → WEquiv u w
+set_option linter.uppercaseLean3 false in
+#align mvqpf.Wequiv MvQPF.WEquiv
+
+theorem recF_eq_of_wequiv (α : TypeVec n) {β : Type _} (u : F (α.append1 β) → β) (x y : q.P.W α) :
```
Comment (alexkeizer) [inline Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean:95]:
```suggestion
theorem recF_eq_of_wEquiv (α : TypeVec n) {β : Type _} (u : F (α.append1 β) → β) (x y : q.P.W α) :
```

File: Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+theorem WEquiv.symm {α : TypeVec n} (x y : q.P.W α) : WEquiv x y → WEquiv y x := by
+  intro h; induction h
+  case ind a f' f₀ f₁ _h ih => exact WEquiv.ind _ _ _ _ ih
+  case abs a₀ f'₀ f₀ a₁ f'₁ f₁ h => exact WEquiv.abs _ _ _ _ _ _ h.symm
+  case trans x y z _e₁ _e₂ ih₁ ih₂ => exact MvQPF.WEquiv.trans _ _ _ ih₂ ih₁
+set_option linter.uppercaseLean3 false in
+#align mvqpf.Wequiv.symm MvQPF.WEquiv.symm
+
+/-- maps every element of the W type to a canonical representative -/
+def wrepr {α : TypeVec n} : q.P.W α → q.P.W α :=
```
Comment (alexkeizer) [inline Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean:137]:
```suggestion
def wRepr {α : TypeVec n} : q.P.W α → q.P.W α :=
```
This is a minor suggestion, feel free to ignore it, but to it feels `wRepr` should be two words, so `Repr` capitalized. If you do make this change, also change it down below

File: Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+  apply q.P.w_ind _ x; intro a f' f ih
+  apply WEquiv.trans _ (q.P.wMk' (appendFun id wrepr <$$> ⟨a, q.P.appendContents f' f⟩))
+  · apply WEquiv.abs'
+    rw [wrepr_wMk, q.P.wDest'_wMk', q.P.wDest'_wMk', abs_repr]
+  rw [q.P.map_eq, MvPFunctor.wMk', appendFun_comp_splitFun, id_comp]
+  apply WEquiv.ind; exact ih
+set_option linter.uppercaseLean3 false in
+#align mvqpf.Wrepr_equiv MvQPF.wrepr_equiv
+
+theorem wequiv_map {α β : TypeVec n} (g : α ⟹ β) (x y : q.P.W α) :
```
Comment (alexkeizer) [inline Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean:160]:
```suggestion
theorem wEquiv_map {α β : TypeVec n} (g : α ⟹ β) (x y : q.P.W α) :
```

File: Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean (hunk 2 above)
Comment (arienmalec) [reply Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean:137]: I think not b/c it's not a `Repr` in the sense of the typeclss

File: Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean (hunk 1 above)
Comment (arienmalec) [reply Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean]: I think `of_WEquiv` is better here and elsewhere actually. The main type is `WEquiv`

Comment (alexkeizer) [reply Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean]: The [https://github.com/leanprover-community/mathlib4/wiki/Porting-wiki#naming-convention](convention) says:
> When something named with UpperCamelCase is part of something named with snake_case, it is referenced in lowerCamelCase

So, indeed the main type is `WEquiv`, but we spell it `wEquiv` when it's part of a snake_case theorem. (Admittedly, I don't think we are being very consistent with this part of the convention)

Comment (arienmalec) [reply Mathlib/Data/QPF/Multivariate/Constructions/Fix.lean]: you're correct... `WEquiv` is `FooEquiv` which would be `of_fooEquiv`

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**